Sources/NIO/Interfaces.swift

```diff
@@ -206,6 +227,57 @@ extension NIONetworkDevice {
         /// This constructor will fail if NIO does not understand the format of the underlying
         /// socket address family. This is quite common: for example, Linux will return AF_PACKET
         /// addressed interfaces on most platforms, which NIO does not currently understand.
+#if os(Windows)
+        internal init?(_ pAdapter: UnsafeMutablePointer<IP_ADAPTER_ADDRESSES>,
+                       _ pAddress: UnsafeMutablePointer<IP_ADAPTER_UNICAST_ADDRESS>) {
+            self.name = String(decodingCString: pAdapter.pointee.FriendlyName,
+                               as: UTF16.self)
+            self.address = pAddress.pointee.Address.lpSockaddr.convert()
+
+            let v4mask: (UINT8) -> SocketAddress? = { _ in
+                var buffer: [CChar] =
+                    Array<CChar>(repeating: 0, count: Int(INET_ADDRSTRLEN))
+                var mask: sockaddr_in = sockaddr_in)
+                mask.sin_family = ADDRESS_FAMILY(BSDSocket.AF_INET)
+                _ = buffer.withUnsafeMutableBufferPointer {
+                    try! BSDSocket.inet_ntop(af: BSDSocket.AF_INET, src: &mask,
+                                             dst: $0.baseAddress!,
+                                             size: INET_ADDRSTRLEN)
+                }
+                return SocketAddress(mask, host: mask.addressDescription())
+            }
+            let v6mask: (UINT8) -> SocketAddress? = { _ in
```

Comment:
Can we get TODOs on these two functions that note that they don't _actually_ calculate the mask? It took me quite a while of looking at this code to note that they are skipping the calculation.

Reply:
I don't think its a TODO item, its just a NOTE right?  These functions are fed the prefix which is then converted to a netmask via `inet_ntop`.

Reply:
These functions explicitly do nothing with the prefix. To be clear, exactly what happens here is:

1. Allocate an empty buffer to write the address into.
2. Create an empty sockaddr of the appropriate family. (initialized to zeros, aka the all-zero address)
3. Call `inet_ntop` to convert the empty address we just allocated to a string in the empty buffer
4. Create a SocketAddress from the empty address, ignoring the buffer.

This means that a) the buffer does nothing, and b) the mask is always the all-zero address. The prefix is never used in these functions, so there is no way it can be involved in creating the netmask.

Reply:
Ah, right.  Okay, lets make this a TODO for now, getting everything building and then fixing is going to be easier here as it will allow running.